workflow/react-project/src/components/common/Modal/ModalContent/RandomCocktail/index.jsx

```diff
@@ -0,0 +1,56 @@
+import { useSelector } from 'react-redux';
+
+import { faStar } from '@fortawesome/free-regular-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import './style.less';
+
+const RandomCocktail = () => {
+  const randomCocktailStatus = useSelector(state => state.randomCocktail?.status);
+  const randomCocktailDetails = useSelector(state => state.randomCocktail?.details)?.drinks?.[0];
+  const randomCocktail = Object.entries(randomCocktailDetails || {});
+  const cocktailName = randomCocktailDetails?.strDrink;
+  const cocktailPicture = randomCocktailDetails?.strDrinkThumb;
+  const instructions = randomCocktailDetails?.strInstructions;
+  const ingredients = randomCocktail?.filter(item => item[0].includes('strIngredient') && item[1]);
+  const quantity = randomCocktail?.filter(item => item[0].includes('strMeasure') && item[1]);
```

Comment:
you can modify ingredients and quantity when you'll modify data before writing to the store.
Why: because  RandomCocktail is a function component and like ordinary function it will be executed each component rerender. So like ordinary function every execution it will filter the same `randomCocktail` array.
But in reducer we will do it one time - after getting data.

it's not really useful to store ingredients and quantity in [key, value] way, because you don't need this `strIngredient` key at all, and you need to get value by index.
Better to make go through `randomCocktail` by  reducer method and create array like this: 
```
[
{
 ingredient: "Strawberries",
 quantity: 2,
mesure: "shots"
}
]
```
this array of ingredients will be more handy for mapping.